### PR TITLE
Add fix.sh entry for GtkEntry::icon-press/-release

### PR DESCRIPTION
**Pull request by nt8r**
This addresses gtk-rs#1051.

### CODE REVIEW

**Comment by sdroege** [comment]
Can you also run the script and put the changes into a second commit?

@GuillaumeGomez @EPashkin What do you think? Also this can't be done via the `Gir.toml`, right?

File: fix.sh [hunk 1]
```diff
@@ -12,6 +12,12 @@ xmlstarlet ed -P -L \
 	-u '//*[@glib:error-domain="g-option-context-error-quark"]/@glib:error-domain' -v g-option-error-quark \
 	GLib-2.0.gir
 
+# incorrectly assumes GdkEventButton when other variants may be passed
+xmlstarlet ed -P -L \
+	-u '//glib:signal[@name="icon-press"]//_:parameter[@name="event"]/_:type[@name="Gdk.EventButton"]/@name' -v "Gdk.Event" \
+	-u '//glib:signal[@name="icon-release"]//_:parameter[@name="event"]/_:type[@name="Gdk.EventButton"]/@name' -v "Gdk.Event" \
```
Comment (sdroege) [inline fix.sh:18]: Might be useful to add another condition at the beginning so this only ever goes into `GtkEntry` and not anything else that might have the same signal name in the future.

Comment (nt8r) [reply fix.sh]: Done.

**Comment by nt8r** [comment]
Second commit with changes added. Running the script also permuted some fields in two lines of `JavascriptCore-4.0.gir`, but I didn't add those changes because they aren't related:
```
diff --git a/JavaScriptCore-4.0.gir b/JavaScriptCore-4.0.gir
index 9873cca..b8f9a6a 100644
--- a/JavaScriptCore-4.0.gir
+++ b/JavaScriptCore-4.0.gir
@@ -726,7 +726,7 @@ with jsc_value_new_object() that receives the copy as the instance parameter.</d
             <doc xml:space="preserve">the #GType of the property value</doc>
             <type name="GType" c:type="GType"/>
           </parameter>
-          <parameter name="getter" transfer-ownership="none" nullable="1" allow-none="1" scope="notified" closure="4" destroy="5">
+          <parameter name="getter" transfer-ownership="none" nullable="1" allow-none="1" destroy="5" scope="notified" closure="4">
             <doc xml:space="preserve">a #GCallback to be called to get the property value</doc>
             <type name="PropertyGetter" c:type="JSCPropertyGetter"/>
           </parameter>
@@ -2579,7 +2579,7 @@ with jsc_value_new_object() that receives the copy as instance parameter.</doc>
             <doc xml:space="preserve">the #GType of the property</doc>
             <type name="GType" c:type="GType"/>
           </parameter>
-          <parameter name="getter" transfer-ownership="none" nullable="1" allow-none="1" scope="notified" closure="5" destroy="6">
+          <parameter name="getter" transfer-ownership="none" nullable="1" allow-none="1" destroy="6" scope="notified" closure="5">
             <doc xml:space="preserve">a #GCallback to be called to get the property value</doc>
             <type name="Getter" c:type="JSCGetter"/>
           </parameter>
```

**Review by sdroege — APPROVED** [review]
lgtm! @EPashkin @GuillaumeGomez ?

**Review by GuillaumeGomez — APPROVED** [review]
:+1:

**Comment by sdroege** [comment]
@nt8r Do you also want to submit a PR for gtk to regenerate it? And if you feel like it, you could also add the fix for https://github.com/gtk-rs/gtk/issues/1050 at the same time (just requires a `nullable=true` configuration in the `Gir.toml`, but I can also do that in the next days otherwise).

**Pull request merged**